File: JeyahShopAPI/src/main/java/com/jeyah/jeyahshopapi/product/ProductResponse.java
package com.jeyah.jeyahshopapi.product;

import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductResponse {

    private Integer id;
    private String name;
    private Integer price;
    private String description;
    private Integer categoryId;
    private Integer stockQuantity;
    private List<String> imageUrls;
    private double rate;
    private boolean available;


}

File: JeyahShopUI/src/app/shared/models/product-response.ts
// src/app/shared/models/product-response.ts

export interface ProductResponse {
  id: number;
  name: string;
  price: number;
  description: string;
  categoryId: number;
  stockQuantity: number;
  imageUrls: string[];
  rate: number;
  available: boolean;
}
